### PR TITLE
Bug/quote dependency scripts

### PR DESCRIPTION
**Pull request by cm421**
### Intent

Allow the dependency scripts to be run in a directory that has spaces at some point in the path, and to be run on a fresh install of Ubuntu.

### Approach

Updated Ubuntu scripts to include dependencies that are needed in the scrips, but not included in a fresh Ubuntu install. Added quotes to scripts that refer to the working directory to account for any spaces in the path.

### Automated Tests

N/A

### QA Notes

I've tested these on fresh Ubuntu focal/bionic/stretch VMs, which is where I was having the issues.

### Checklist

- [ x ] If this PR adds a new feature, or fixes a bug in a previously released version, it includes an entry in `NEWS.md` 
- [ x ] If this PR adds or changes UI, the updated UI meets [accessibility standards](https://github.com/rstudio/rstudio/wiki/Accessibility)
- [ x ] A reviewer is assigned to this PR (if unsure who to assign, check Area Owners list)
- [ x ] This PR passes all local unit tests

### CODE REVIEW

File: dependencies/common/install-npm-dependencies [hunk 1]
```diff
@@ -85,7 +85,7 @@ if [ ! -x "$(command -v yarn)" ]; then
   rm $YARN_INSTALL_SCRIPT
   INSTALL_PATH="${HOME}/.yarn/bin:${HOME}/.config/yarn/global/node_modules/.bin:${INSTALL_PATH}"
 else
-  YARN_VERSION=`PATH=${INSTALL_PATH} yarn --version`
+  YARN_VERSION="PATH=${INSTALL_PATH} yarn --version"
```
Comment (kevinushey) [inline dependencies/common/install-npm-dependencies:88]: This doesn't look correct to me. Do you want to quote `${INSTALL_PATH}` within the call? E.g.

```suggestion
  YARN_VERSION="$(PATH="${INSTALL_PATH}" yarn --version)"
```

You can also use [shellcheck](https://github.com/koalaman/shellcheck) which is pretty good at spotting these.

**Review by kevinushey — CHANGES_REQUESTED** [review]

File: dependencies/common/install-npm-dependencies (hunk 1 above)
Comment (jeffvroom) [reply dependencies/common/install-npm-dependencies:88]: I don't think you want to get rid of the backquotes, how about:

```suggestion
YARN_VERSION=`$(PATH="${INSTALL_PATH}" yarn --version)`
```

Comment (cm421) [reply dependencies/common/install-npm-dependencies:88]: Oops! Great catch

File: dependencies/common/install-npm-dependencies [hunk 2]
```diff
@@ -85,7 +85,7 @@ if [ ! -x "$(command -v yarn)" ]; then
   rm $YARN_INSTALL_SCRIPT
   INSTALL_PATH="${HOME}/.yarn/bin:${HOME}/.config/yarn/global/node_modules/.bin:${INSTALL_PATH}"
 else
-  YARN_VERSION=`PATH=${INSTALL_PATH} yarn --version`
+YARN_VERSION=`$(PATH="${INSTALL_PATH}" yarn --version)`
```
Comment (kevinushey) [inline dependencies/common/install-npm-dependencies:88]:
```suggestion
YARN_VERSION=$(PATH="${INSTALL_PATH}" yarn --version)
```

I think you want to either use backticks or `$()` for shell expansion, not both.

File: dependencies/common/install-npm-dependencies (hunk 1 above)
Comment (MariaSemple) [reply dependencies/common/install-npm-dependencies]: I don't think you want backquotes and `$()`, they do the same thing. It should either be `$(PATH="${INSTALL_PATH}" yarn --version)` or `<backtick>PATH="${INSTALL_PATH}" yarn --version)<backtick>`

**Review by kevinushey — APPROVED** [review]

**Pull request merged**